### ISSUE TITLE
[20.3.19] Segmentation fault in MOVE PARTITION

### ISSUE THREAD

**Issue by UnamedRus** [ClickHouse/ClickHouse#16274]
**Describe the bug**
If we move data between replicated tables, and both of them pointing on the same path in zookeeper segmentation violation occurs.

**How to reproduce**
Clickhouse server 20.3.19
```
CREATE TABLE table_repl
(
   `number` UInt32
)
ENGINE = ReplicatedMergeTree('/clickhouse/test/tables/test/table_repl', 'repl')
PARTITION BY intDiv(number, 1000)
ORDER BY number;
INSERT INTO table_repl SELECT * FROM numbers(1000,2000);

### in zkCli.sh
deleteall  /clickhouse/test/tables/test/table_repl

SYSTEM RESTART REPLICA table_repl;

CREATE TABLE table_repl_rec AS default.table_repl;

ALTER TABLE table_repl MOVE PARTITION 1 TO TABLE table_repl_rec;
ALTER TABLE table_repl MOVE PARTITION 2 TO TABLE table_repl_rec;
```
**Error message and/or stacktrace**
```
2020.10.22 23:56:33.199948 [ 19255 ] {} <Fatal> BaseDaemon: ########################################
2020.10.22 23:56:33.200015 [ 19255 ] {} <Fatal> BaseDaemon: (version 20.3.19.4 (official build)) (from thread 19221) (query_id: e7c3afe9-31a5-4eca-a4be-39ce2343b221) Received signal Segmentation fault (11).
2020.10.22 23:56:33.200039 [ 19255 ] {} <Fatal> BaseDaemon: Address: 0x38 Access: read. Address not mapped to object.
2020.10.22 23:56:33.200078 [ 19255 ] {} <Fatal> BaseDaemon: Stack trace: 0xcff8730 0xcc9d5a7 0xccbaa1d 0xc5e6c8e 0xca74ef8 0xca77b31 0x86514e9 0x86524d0 0xd8d80bb 0xd8d853d 0xe22a7d7 0xe2265cc 0xe227f6d 0x7fc6e3a67609 0x7fc6e3cfd293
2020.10.22 23:56:33.200780 [ 19255 ] {} <Fatal> BaseDaemon: 3. DB::ReplicatedMergeTreeQueue::removePartProducingOpsInRange(std::__1::shared_ptr<zkutil::ZooKeeper>, DB::MergeTreePartInfo const&, DB::ReplicatedMergeTreeLogEntryData const&) @ 0xcff8730 in /usr/lib/debug/usr/bin/clickhouse
2020.10.22 23:56:33.201549 [ 19255 ] {} <Fatal> BaseDaemon: 4. DB::StorageReplicatedMergeTree::movePartitionToTable(std::__1::shared_ptr<DB::IStorage> const&, std::__1::shared_ptr<DB::IAST> const&, DB::Context const&) @ 0xcc9d5a7 in /usr/lib/debug/usr/bin/clickhouse
2020.10.22 23:56:33.202239 [ 19255 ] {} <Fatal> BaseDaemon: 5. DB::StorageReplicatedMergeTree::alterPartition(std::__1::shared_ptr<DB::IAST> const&, DB::PartitionCommands const&, DB::Context const&) @ 0xccbaa1d in /usr/lib/debug/usr/bin/clickhouse
2020.10.22 23:56:33.202829 [ 19255 ] {} <Fatal> BaseDaemon: 6. DB::InterpreterAlterQuery::execute() @ 0xc5e6c8e in /usr/lib/debug/usr/bin/clickhouse
2020.10.22 23:56:33.203460 [ 19255 ] {} <Fatal> BaseDaemon: 7. ? @ 0xca74ef8 in /usr/lib/debug/usr/bin/clickhouse
2020.10.22 23:56:33.204155 [ 19255 ] {} <Fatal> BaseDaemon: 8. DB::executeQuery(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, DB::Context&, bool, DB::QueryProcessingStage::Enum, bool, bool) @ 0xca77b31 in /usr/lib/debug/usr/bin/clickhouse
2020.10.22 23:56:33.204851 [ 19255 ] {} <Fatal> BaseDaemon: 9. DB::TCPHandler::runImpl() @ 0x86514e9 in /usr/lib/debug/usr/bin/clickhouse
2020.10.22 23:56:33.205580 [ 19255 ] {} <Fatal> BaseDaemon: 10. DB::TCPHandler::run() @ 0x86524d0 in /usr/lib/debug/usr/bin/clickhouse
2020.10.22 23:56:33.206294 [ 19255 ] {} <Fatal> BaseDaemon: 11. Poco::Net::TCPServerConnection::start() @ 0xd8d80bb in /usr/lib/debug/usr/bin/clickhouse
2020.10.22 23:56:33.206972 [ 19255 ] {} <Fatal> BaseDaemon: 12. Poco::Net::TCPServerDispatcher::run() @ 0xd8d853d in /usr/lib/debug/usr/bin/clickhouse
2020.10.22 23:56:33.207635 [ 19255 ] {} <Fatal> BaseDaemon: 13. Poco::PooledThread::run() @ 0xe22a7d7 in /usr/lib/debug/usr/bin/clickhouse
2020.10.22 23:56:33.208303 [ 19255 ] {} <Fatal> BaseDaemon: 14. Poco::ThreadImpl::runnableEntry(void*) @ 0xe2265cc in /usr/lib/debug/usr/bin/clickhouse
2020.10.22 23:56:33.208974 [ 19255 ] {} <Fatal> BaseDaemon: 15. ? @ 0xe227f6d in /usr/lib/debug/usr/bin/clickhouse
2020.10.22 23:56:33.209019 [ 19255 ] {} <Fatal> BaseDaemon: 16. start_thread @ 0x9609 in /usr/lib/x86_64-linux-gnu/libpthread-2.31.so
2020.10.22 23:56:33.209822 [ 19255 ] {} <Fatal> BaseDaemon: 17. clone @ 0x122293 in /usr/lib/x86_64-linux-gnu/libc-2.31.so
```

**Additional context**
In 20.8 MOVE PARTITION working correctly.

```
ALTER TABLE table_repl_rec ATTACH PARTITION 1 FROM table_repl;
ALTER TABLE table_repl_rec ATTACH PARTITION 2 FROM table_repl;
```
In 20.3 ATTACH PARTITION working correctly too.

**Comment by alesapin** [comment]
But why anybody needs to create two replicated tables on the same path in ZooKeeper? Looks like a hacky tick.

**Comment by UnamedRus** [comment]
A simpler recovery from lost zookeeper, almost all interactions are being done via clickhouse queries (you would need only to rm the ATTACH script in metadata dir for the old table).

**Comment by alexey-milovidov** [comment]
> Additional context
In 20.8 MOVE PARTITION working correctly.

It means that the issue is already fixed in recent versions?

But we are not going to search for the fix and backport in to version 20.3, because it happens only in specific scenario that is only possible with manual intervention.
